app/scripts/services/import-service.js

```diff
@@ -183,8 +184,36 @@
             if (option === importServiceConflictModal.REPLACE_FILE) {
               var path = ramlRepository.join(directory.path, name);
               var file = ramlRepository.getByPath(path);
+              var promise;
 
-              file.doc.setValue(contents);
+              if (file) {
+                // Load the file if not loaded.
+                promise = file.loaded ? $q.when(file) : ramlRepository.loadFile({path: path});
```

Comment:
that's one reason, and the other one is that I'm setting the file as dirty as it's not being saved and the user needs to see that feedback not to lose changes. Then, in the cases where the file was not open, when the user clicks the file in the file browser, if it's not marked as loaded, it's retrieved again from the repository and the new contents are overwritten and the dirty check is lost. That's why I'm loading the original contents here, so that the file is loaded and the dirty check works correctly.

Does it make sense?